README.md

```diff
@@ -178,36 +189,106 @@ store.firestore.get({ collection: 'cities' }),
 // store.firestore.get({ collection: 'cities', doc: 'SF' }), // doc
 ```
 
-##### set
+##### mutate
+
+**Single Document** - merge multiple changes for a single document
+
+```js
+store.firestore.mutate({
+  collection: 'cities',
+  doc: 'SF',
+  data: {
+    name: 'San Francisco',
+  },
+});
+```
+
+**Batch Updates** - run multiple changes as a single batch. Arrays larger
+then 500 will be broken into multiple batches when sent to firestore, then
+recombined and flattened when returned in the promise.
+
+```js
+store.firestore.mutate([{
+  collection: 'cities',
+  doc: 'SF',
+  data: {
+    name: 'San Francisco',
+  },
+  {
+    collection: 'cities',
+    doc: 'Miami',
+    data: {
+      name: 'Miami',
+    },
+  },
+}]);
+```
+
+**Firestore Transactions** - Read keys are used in dependency injected into
+the write functions. Static writes are also supported. A read that does not
+have a collection/doc value will be use to provide and dependency inject
+state data to the write function.
+
+```js
+store.firestore
+  .mutate({
+    reads: {
+      sanFrancisco: { collection: 'cities', doc: 'SF' },
+    },
+    writes: [
+      ({ sanFrancisco }) => ({
+        collection: 'cities',
+        doc: 'SF',
+        data: {
+          population: sanFrancisco.population + 1,
+        },
+      }),
+    ],
+  })
+  .then((result) => {
+    // TRANSACTION_SUCCESS action dispatched
+    console.log('Transaction success!');
+  })
+  .catch((err) => {
+    // TRANSACTION_FAILURE action dispatched
+    console.log('Transaction failure:', err);
+  });
+```
+
+#### Old API Actions
+
+The older API is a smaller wrapper around Firestore. It exposes most
+functionality but is not compatible with the cache reducer. All changes
+sent to the old API must wait for the data to complete a round-trip to the
+Firestore database in Google Cloud before it will be shown to the user.
+
+For changes that are reflected immeadiately to the user, see the mutate API.
+
+##### set - old api
```

Comment:
Thanks for updating the README with docs!

I would say that calling this this "old api" leads me to think we should be doing a major version bump here, when I don't really think that is necessary since it isn't breaking this existing function, it just doesn't work with the new stuff.

Since you already put so much work in, I can clean up that wording as part of the release process